src/Shared/UnitTests/FileMatcher_Tests.cs

```diff
@@ -915,7 +915,7 @@ private int GetMatchingFiles(string[] candidates, string path, string pattern, H
                         string candidateDirectoryName = "";
                         if (normalizedCandidate.IndexOfAny(FileMatcher.directorySeparatorCharacters) != -1)
                         {
-                            candidateDirectoryName = Path.GetDirectoryName(normalizedCandidate) + @"\";
+                            candidateDirectoryName = Path.GetDirectoryName(normalizedCandidate);
```

Comment:
FileMatcher_Tests were not picking up the bug because the mock file system was also appending spurious slashes :)